test/labeling/model/test_label_model.py

```diff
@@ -79,6 +115,115 @@ def test_augmented_L_construction(self):
+        np.testing.assert_array_almost_equal(accs, np.array([0.7, 0.825]))
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone())
+        accs = label_model.get_accuracies(probs=None)
+        np.testing.assert_array_almost_equal(accs, np.array([0.5, 0.355]))
+
+    def test_build_mask(self):
+
+        L = np.array([[1, 2, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        true_mask = np.array(
+            [
+                [0, 0, 1, 1, 1, 1],
+                [0, 0, 1, 1, 1, 1],
+                [1, 1, 0, 0, 1, 1],
+                [1, 1, 0, 0, 1, 1],
+                [1, 1, 1, 1, 0, 0],
+                [1, 1, 1, 1, 0, 0],
+            ]
+        )
+
+        mask = label_model.mask.numpy()
+        np.testing.assert_array_equal(mask, true_mask)
+
+    def test_init_params(self):
+        L = np.array([[1, 2, 1], [1, 0, 1]])
+        label_model = self._set_up_model(L, class_balance=[0.6, 0.4])
+
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.8750], [1.0, 0.0], [0.0, 0.0]]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+        label_model._set_class_balance(class_balance=[0.3, 0.7], Y_dev=None)
+        label_model._init_params()
+
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.5], [1.0, 0.0], [0.0, 0.0]]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+    def test_predict_proba(self):
+        L = np.array([[1, 2, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone())
+        probs = label_model.predict_proba(L)
+
+        true_probs = np.array([[0.99, 0.01], [0.99, 0.01]])
+        np.testing.assert_array_almost_equal(probs, true_probs)
+
+    def test_loss(self):
+        L = np.array([[1, 0, 1], [1, 2, 0]])
+        label_model = self._set_up_model(L)
+        label_model._get_augmented_label_matrix(L, higher_order=True)
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone() + 0.05)
+        self.assertAlmostEqual(
+            label_model.loss_l2(l2=1.0).detach().numpy().ravel()[0], 0.03
+        )
+        self.assertAlmostEqual(
+            label_model.loss_l2(l2=np.ones(6)).detach().numpy().ravel()[0], 0.03
+        )
+        self.assertAlmostEqual(
+            label_model.loss_mu().detach().numpy().ravel()[0], 0.675, 3
+        )
+
+    def test_loss_decrease(self):
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.train_model(L, n_epochs=1)
+        init_loss = label_model.loss_mu().detach().numpy().ravel()[0]
+
+        label_model.train_model(L, n_epochs=10)
+        next_loss = label_model.loss_mu().detach().numpy().ravel()[0]
+
+        self.assertLessEqual(next_loss, init_loss)
```

Comment:
Repeating myself, but isn't this highly dependent on default hyperparams? Should we hardcode those (even just LR) into the call here in case they change?

Reply:
my bad, ill hardcode the params to be safe